pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -1653,16 +1703,28 @@ type buildListenerOpts struct {
 	needHTTPInspector bool
 }
 
-func buildHTTPConnectionManager(node *model.Proxy, env *model.Environment, httpOpts *httpListenerOpts,
+func buildHTTPConnectionManager(pluginParams *plugin.InputParams, env *model.Environment, httpOpts *httpListenerOpts,
 	httpFilters []*http_conn.HttpFilter) *http_conn.HttpConnectionManager {
-
 	filters := make([]*http_conn.HttpFilter, len(httpFilters))
 	copy(filters, httpFilters)
 
 	if httpOpts.addGRPCWebFilter {
 		filters = append(filters, &http_conn.HttpFilter{Name: wellknown.GRPCWeb})
 	}
 
+	// append ALPN HTTP filter in HTTP connection manager for outbound listener only.
+	if pluginParams.ListenerCategory == networking.EnvoyFilter_SIDECAR_OUTBOUND ||
+		pluginParams.DeprecatedListenerCategory == networking.EnvoyFilter_DeprecatedListenerMatch_SIDECAR_OUTBOUND {
+		filters = append(filters, &http_conn.HttpFilter{
+			Name: AlpnFilterName,
+			ConfigType: &http_conn.HttpFilter_TypedConfig{
```

Comment:
perf: make this HttpFilter a constant

Reply:
A variable inside `var`?

Reply:
We might need to move all other filters from inline definition to some global area.